R/module_filter_manager.R

```diff
@@ -107,109 +117,85 @@ filter_manager_srv <- function(id, filtered_data_list, filter) {
     }
     filtered_data_list <- flatten_nested(filtered_data_list)
 
-    # global list of slices (all available teal_slice)
-    slices_global <- reactiveVal(filter)
+    # # flatten nested alternative
+    # filtered_data_list <- unlist(filtered_data_list)
+    # names(filtered_data_list) <- sub("(.+)\\.(.+$)", "\\2", names(filtered_data_list))
 
-    # create a reactive map between modules and filters
-    slices_map <- sapply(
-      names(filtered_data_list),
-      function(module_name) {
-        shiny::reactiveVal(
-          unlist(attr(filter, "mapping")[c(module_name, "global_filters")], use.names = FALSE)
-        )
-      }
-    )
+    # Create mapping fo filters to modules in matrix form (presented as data.frame).
+    mapping_matrix <- reactive({
+      mapping_ragged <- lapply(filtered_data_list, function(x) slices_field(x$get_filter_state(), "id"))
+      all_names <- slices_field(slices_global(), "id")
+      mapping_smooth <- lapply(mapping_ragged, is.element, el = all_names)
+      as.data.frame(mapping_smooth, row.names = all_names)
+    })
```

Comment:
mapping_matrix doesn't handle module names properly. Spaces are turned into dots and thus restore process doesn't work

Reply:
This should be solved by adding `check.names = FALSE`.